website/main/content/blog/2025-04-14-data-availability.md

```diff
@@ -0,0 +1,142 @@
+---
+title: The data availability problem
+date: 2025-04-14
+draft: false
+description: How to ensure that blocks and segments are available in a shard when you need them.
+tags: [status-update, consensus]
+authors: [adlrocha]
+---
+
+This week has been another good week of progress. I finally have a good idea of how shard archiving
+should work in the happy path, and I've started writing a low-level spec for it (that I am hoping to
+push to this repo soon). Unfortunately, there is still a slight gap in the spec that we need to fill
+before we can move forward: the _data availability problem_.
+
+<!--more-->
+
+## Current state of sharded archiving design
+
+High-level, this is how the end-to-end of sharded archiving currently looks like:
+
+- _Block Creation (L2 Child):_ New blocks (`blk1`, `blk2`, etc.) are created normally in the child
+  shard. As soon as they are created, their headers (or block hashes) are immediately committed to
+  the parent chain (L1 child).
+- _Parent Shard Tracking (L1 Child):_ The parent shard keeps track in a block buffer of all the
+  headers (or hashes) from the child shard blocks that have been committed, creating in this way an
+  intermediate block history of all its child shards until a new segment from the shard arrives.
+- _Block History Management (L2 Child):_ When the history buffer in a shard accumulates enough
+  blocks, a new segment is created in the shard. This triggers the creation of a super segment
+  (`super_segment_L2_1`) which includes information about all segments and commitments added to the
+  history and that is correspondingly submitted to the parent for commitment. It is worth nothing
+  that super segment do not propagate the whole segment, but aggregate commitments for the segments
+  that can be used to verify them in the upper layers of the system.
+- _Recursive operation (L1 Child):_ The protocol is recursive, so in the same way that the L1's
+  child shard was committing blocks to the parent as they were being created, it will do the same
+  with its own parent (the beacon chain).
+- _Super-Segment Commitment (L1 Child):_ When a super-segment from a child (`super_segment_L2_1`) is
+  committed, it is added to the shard's history buffer along with other local segments that may have
+  been created, triggering the clearing of the block buffer for child shards. Once a segment is
+  committed in the parent, there is no need to keep raw blocks in the buffer anymore as they are
+  explicitly available in the committed segments.
+- _Beacon Chain Integration (L1 Child):_ The beacon chain receives the parent shard's blocks and
```

Comment:
Those shards are not parents relatively to the beacon chain though

Reply:
I see how this phrasing can be confusing. I rephrased it a bit, let me know how it looks now.